test/unit/builtins/map.spec.ts

```diff
@@ -1,95 +1,79 @@
-    );
-
-    expect(result).toBe(true);
+    util.testFunction`
+        let mymap = new Map([["a", "c"],["b", "d"]]);
+         return mymap.has("a") && mymap.has("b");
+    `.expectToMatchJsResult();
 });
 
 test("map iterable constructor map", () => {
-    const result = util.transpileAndExecute(`
+    util.testFunction`
         let mymap = new Map(new Map([["a", "c"],["b", "d"]]));
         return mymap.has("a") && mymap.has("b");
-    `);
-
-    expect(result).toBe(true);
+    `.expectToMatchJsResult();
 });
 
 test("map clear", () => {
     const mapTS = 'let mymap = new Map([["a", "c"],["b", "d"]]); mymap.clear();';
-    const size = util.transpileAndExecute(mapTS + "return mymap.size;");
-    expect(size).toBe(0);
+    util.testExpression("mymap.size;").setTsHeader(mapTS).expectToMatchJsResult();
 
-    const contains = util.transpileAndExecute(mapTS + 'return !mymap.has("a") && !mymap.has("b");');
-    expect(contains).toBe(true);
+    util.testExpression('!mymap.has("a") && !mymap.has("b");').setTsHeader(mapTS).expectToMatchJsResult();
 });
 
 test("map delete", () => {
-    const mapTS = 'let mymap = new Map([["a", "c"],["b", "d"]]); mymap.delete("a");';
-    const contains = util.transpileAndExecute(mapTS + 'return mymap.has("b") && !mymap.has("a");');
-    expect(contains).toBe(true);
+    util.testFunction`
+        let mymap = new Map([["a", "c"],["b", "d"]]);
+        mymap.delete("a");
+        return mymap.has("b") && !mymap.has("a");
+    `.expectToMatchJsResult();
 });
 
 test("map entries", () => {
-    const result = util.transpileAndExecute(
-        `let mymap = new Map([[5, 2],[6, 3],[7, 4]]);
+    util.testFunction`
+        let mymap = new Map([[5, 2],[6, 3],[7, 4]]);
         let count = 0;
         for (const [key, value] of mymap.entries()) { count += key + value; }
-        return count;`
-    );
-    expect(result).toBe(27);
+        return count;
+    `.expectToMatchJsResult();
 });
 
 test("map foreach", () => {
-    const result = util.transpileAndExecute(
+    util.testFunction(
         `let mymap = new Map([["a", 2],["b", 3],["c", 4]]);
         let count = 0;
         mymap.forEach(i => count += i);
         return count;`
-    );
-
-    expect(result).toBe(9);
+    ).expectToMatchJsResult();
 });
 
 test("map foreach keys", () => {
-    const result = util.transpileAndExecute(
-        `let mymap = new Map([[5, 2],[6, 3],[7, 4]]);
+    util.testFunction`
+        let mymap = new Map([[5, 2],[6, 3],[7, 4]]);
         let count = 0;
         mymap.forEach((value, key) => { count += key; });
-        return count;`
-    );
-
-    expect(result).toBe(18);
+        return count;
+    `.expectToMatchJsResult();
 });
 
 test("map get", () => {
-    const result = util.transpileAndExecute('let mymap = new Map([["a", "c"],["b", "d"]]); return mymap.get("a");');
-
-    expect(result).toBe("c");
+    util.testFunction('let mymap = new Map([["a", "c"],["b", "d"]]); return mymap.get("a");').expectToMatchJsResult();
```

Comment:
For this test (and the 4 below), could you also use the template string syntax with newlines (like in the other tests)